File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int n=nums.size();
        vector<int> vleft(n);
        vector<int> vright(n);
        vector<int> res(n);

        vleft[0]=1;
        vright[n-1]=1;
        for(int i=1;i<n;i++){
            vleft[i]=nums[i-1]*vleft[i-1];
        }
        for(int i=n-2;i>=0;i--){
            vright[i] = vright[i+1]*nums[i+1];
        }
        for(int i=0;i<n;i++){
            res[i] = vleft[i]*vright[i];
        }
        return res;
    }
};

File: Data Structures & Algorithms/combination-target-sum/submission-0.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        nums.sort()
        stk = []
        res = []
        
        def solve(ind, stk, csum):
            if ind==len(nums):
                return
            if csum==target:
                if stk not in res:
                    res.append(stk.copy())
                return
            if nums[ind]>target-csum:
                return
            if csum<target:
                solve(ind+1,stk,csum)
                stk.append(nums[ind])
                solve(ind, stk, csum+nums[ind])
                solve(ind+1, stk, csum+nums[ind])
                stk.pop()
        
        solve(0,[],0)

        return res